### ISSUE TITLE
Tap Comment Output is not valid according to Junit schema

### ISSUE THREAD

**Issue by ketumbra** [dhershman1/tap-junit#37]
**Describe the bug**
When accepting tap input (from mocha) with tap comments `#`, the xml output contains the comment as string content in the `testcase` element (in this case, summarising number of tests, passes and fails). This is not valid according to https://github.com/junit-team/junit5/blob/43638eb6a870e0d6c49224053dfeb39dcf0ef33f/platform-tests/src/test/resources/jenkins-junit.xsd and is rejected by GitLab CI and Jenkins.
e.g.
```xml
<testsuites tests="1" name="junit.xml" failures="0">
<testsuite name="tests" tests="1" skipped="0" failures="0">
<testcase name="test1" id="1"># tests 1
# pass 1
# fail 0
</testcase>
</testsuite>
</testsuites>
```

This appears similar to #32 - comments in the tap file are being added as content of the `testcase` element but this is invalid.

**To Reproduce**
1. set mocha's reporter to `tap`
1. `nyc mocha test.spec.ts | tap-junit --classname tests --name junit.xml -p `

**Expected behavior**
XML is valid according to the above schema.

**Command Line Usage**
If you're able please specify the command line format you used for tap-junit for example:
```cli
nyc mocha test.spec.ts | tap-junit --output out --classname tests --name junit.xml -p && nyc report
```

**Desktop (please complete the following information):**
 - OS: alpine linux x64
 - Node Version: 14
 - Version 4.2.0

**Additional context**
Gitlab rejects xml with text children of the `testcase` node. It doesn't seem to mind the additional `id` attribute on the same element even though this is also invalid according to the above schema.

**Comment by dhershman1** [comment]
First off, I've been looking for a junit 5 schema. I can't believe I didn't think to check their github. This will help standardize the overall output of tap-junit, so thank you for that!

I'll go through this schema file and revamp the output again to match it more closely. The hardest part of this whole thing is it seems every test runner outputs a different TAP style. So I'll have to play with it.

If you don't mind, could you share the TAP output mocha gives you I could add to my tests? 

Thanks!

**Comment by dhershman1** [comment]
Update: I have not forgotten about this, I recently started a new job so that has sort of consumed most of my time

**Comment by dhershman1** [comment]
Merged and released v5.0.0 of tap-junit which should follow the spec more closely, if you notice anything out of line please open another issue and let me know!
